### ISSUE TITLE
Discord role sync problem

### ISSUE THREAD

**Issue by TCMBazsi** [DiscordSRV/DiscordSRV#742]
Hi, I upgraded the plugin to 1.19.0 and for some reason if a player whose minecraft character is linked to the Discord profile disconnects, he will take down the rank that the plugin gives, and is it possible that when a player is offline he does not take the rank off of him on Discord? Here is the[ debug](https://bin.scarsz.me/e5c93d64-dea8-477f-a2a5-19d0fce3f58e#qk3boqHqLpMAT1KtQzvVXQP5s7xf1cq8) link

![unknown](https://user-images.githubusercontent.com/41184885/83443560-3d3a7d80-a44a-11ea-8555-045b41689bd2.png)

**Comment by Vankka** [comment]
Seems like GroupManager doesn't provide groups for offline players for Vault

**Comment by sdcaptains** [comment]
the same problem using luckperms when they are offline the bot will remove the linked role

**Comment by Vankka** [comment]
> the same problem using luckperms when they are offline the bot will remove the linked role

LuckPerms works fine with offline permissions. There is likely a completely different issue.  

---
The issue should be resolved in development builds: https://snapshot.discordsrv.com (groups/roles won't be sync'ed if the permission plugin returns null as the player's groups), let me know how it goes

**Comment by TTRCharlie** [comment]
It now works with the snapshot version, but now, the plugin doesn't write anything in the channeltopic.
Here is the debug link: [https://bin.scarsz.me/100b9c83-1ea1-40c7-9992-72ec80647cf0#JTgresVnc6Ah68OPIZYaiOi6mTCiWGdS](url)

**Comment by Vankka** [comment]
Could you try again with the latest snapshot, https://snapshot.discordsrv.com

**Comment by TTRCharlie** [comment]
> 
> 
> Could you try again with the latest snapshot, https://snapshot.discordsrv.com

I tried with the latest snapshot, but now he only writes in the channel topic that Server is Offline

[[Debug](https://bin.scarsz.me/6fb202f8-cb01-4ffe-9e86-086ef26988a4#CWcwWOkArnJTpbJYO6Hb7J5fEzDq0NSd)]

**Comment by Vankka** [comment]
Made additions changes @ https://github.com/DiscordSRV/DiscordSRV/commit/991347bd4afc27f21d3ab2224dc71e9914b66ca9, please try again

**Comment by TTRCharlie** [comment]
> 
> 
> Made additions changes @ [991347b](https://github.com/DiscordSRV/DiscordSRV/commit/991347bd4afc27f21d3ab2224dc71e9914b66ca9), please try again

Nope, also it writes the same thing now, and in the plugin is doesn't write the Minecraft chat, on Discord.

[[Debug](https://bin.scarsz.me/bcceb3af-a922-4409-89ec-39a7a4056815#egjJZgPfuld7qyqXYmyk1SZzeV4xNfIe)]

**Comment by Vankka** [comment]
Shooould be fixed now

**Comment by TTRCharlie** [comment]
> 
> 
> Shooould be fixed now

Nope, also the same.

[[Debug](https://bin.scarsz.me/cdca902f-7cb5-49a1-b4bd-aeba382928e2#Pyy2pJhGtIY1CfjjaMma67eZFoX9bDpF)]

**Comment by Vankka** [comment]
this time. https://snapshot.discordsrv.com

**Comment by TTRCharlie** [comment]
> 
> 
> this time. https://snapshot.discordsrv.com

Now it's better I mean the plugin writes the Minecraft's chat to Discord and it's can writes the online players, but now it only writes every 10 minutes, regardless of whether it is set to 5 minutes

[[Debug](https://bin.scarsz.me/83e671b8-f111-4add-b3b3-6d44a43d3396#1BnQcVEVP8q9PT714aaH764pcDsWK2Hi)]

**Comment by Vankka** [comment]
That was changed in the last version due to Discord changing the rate limit, https://github.com/DiscordSRV/DiscordSRV/releases/tag/v1.19.0

I assume the offline player group sync issue is fixed now?

**Comment by TTRCharlie** [comment]
> 
> 
> That was changed in the last version due to Discord changing the rate limit, https://github.com/DiscordSRV/DiscordSRV/releases/tag/v1.19.0
> 
> I assume the offline player group sync issue is fixed now?

Yeah, it works now
